src/toil/batchSystems/parasol.py

```diff
@@ -78,17 +76,30 @@ def getUpdatedJob(parasolResultsFile, outputQueue1, outputQueue2):
     
     plus you finally have the command name..
     """
-    parasolResultsFileHandle = open(parasolResultsFile, 'r')
+    resultsFiles = set()
+    resultsFileHandles = []
     while True:
-        line = parasolResultsFileHandle.readline()
-        if line != '':
-            results = line.split()
-            result = int(results[0])
-            jobID = int(results[2])
-            outputQueue1.put(jobID)
-            outputQueue2.put((jobID, result))
-        else:
-            time.sleep(0.01) #Go to sleep to avoid churning
+        #Look for any new results files that have been created, and open them
+        newResultsFiles = set(os.listdir(parasolResultsDir)).difference(resultsFiles)
+        for newFile in newResultsFiles:
+            resultsFiles.add(newFile)
+            newFilePath = os.path.join(parasolResultsDir, newFile)
+            resultsFileHandles.append(open(newFilePath, 'r'))
+
+        for fileHandle in resultsFileHandles:
+            line = fileHandle.readline()
+            if line != '':
+                results = line.split()
+                result = int(results[0])
+                jobID = int(results[2])
+                cpuUsageQueue.put(jobID)
+                updatedJobsQueue.put((jobID, result))
+
+        if not shutdownQueue.empty():
```

Comment:
I see. When I recommended using a None sentinel in the queue I thought that this thread consumed items from a queue. But since this thread doesn't have a queue to read from, it would make more sense to simply use a flag. Here's what I would do: I would make this function a method of the class. Then you wouldn't need to pass the queues in and you could change the while loop in the thread target function to ``while self.running``.